### PR TITLE
GraalJS

### PR DESCRIPTION
**Pull request by rymsha**

### CODE REVIEW

File: modules/script/script-impl/src/test/resources/myapplication/require-test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    console.log(JSON.stringify(xxx, null, 2));
+
+    var bean = __.newBean('com.enonic.xp.script.graal.GraalScriptBean');
+    bean.setSource("Source Name 1");
+    bean.execute();
+    bean.execute(1);
+    bean.execute([1,2,3]);
+    bean.execute("Hello", 1234);
+
+    var mockObj = __.registerMock('/lib/xp/something', {
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapplication/require-test.js:23]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
    var mockObj = __.registerMock("/lib/xp/something", {
```

File: modules/script/script-impl/src/test/resources/myapplication/require-test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const innerLib = require('inner-script');
+
+console.log('I am a require function');
+
+log.info("Print logs with two arguments: %s and %s ", 123.09, "456");
+log.info('info message', new Error('cause'));
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapplication/require-test.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
log.info("info message", new Error('cause'));
```

File: modules/script/script-impl/src/test/resources/myapplication/require-test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const innerLib = require('inner-script');
+
+console.log('I am a require function');
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapplication/require-test.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
console.log("I am a require function");
```

File: modules/script/script-impl/src/test/resources/myapplication/require-test.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+exports.get = function(req) {
+    innerLib.myFunction();
+
+    console.log('GET function is called');
+
+    console.log(JSON.stringify(app, null, 2));
+    console.log(JSON.stringify(xxx, null, 2));
+
+    var bean = __.newBean('com.enonic.xp.script.graal.GraalScriptBean');
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapplication/require-test.js:16]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
    var bean = __.newBean("com.enonic.xp.script.graal.GraalScriptBean");
```

File: modules/script/script-impl/src/test/resources/myapplication/require-test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    bean.execute();
+    bean.execute(1);
+    bean.execute([1,2,3]);
+    bean.execute("Hello", 1234);
+
+    var mockObj = __.registerMock('/lib/xp/something', {
+        a: 1,
+        b: 2
+    });
+    var mock = require('/lib/xp/something');
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapplication/require-test.js:27]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
    var mock = require("/lib/xp/something");
```

File: modules/script/script-impl/src/test/resources/graaljs/java-script.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+(function testJava() {
+     var GraalJavaObjectType = Java.type('com.enonic.xp.script.impl.GraalJavaObject');
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/graaljs/java-script.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
     var GraalJavaObjectType = Java.type("com.enonic.xp.script.impl.GraalJavaObject");
```

File: modules/script/script-impl/src/test/resources/graaljs/function-script.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+input => {
+     if (input === "undefinedValue") {
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/graaljs/function-script.js:2]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
     if (input === 'undefinedValue') {
```

File: modules/script/script-impl/src/test/resources/myapp/script.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+exports.execute = function() {
+    let bean = __.newBean('com.enonic.xp.script.graaljs.impl.ComparingApproaches');
+    bean.setValue("Source Name 1");
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapp/script.js:3]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
    bean.setValue('Source Name 1');
```

File: modules/script/script-impl/src/test/resources/graaljs/function-script.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+        return undefined;
+     } else {
+        return "Hello, " + input;
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/graaljs/function-script.js:5]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
        return 'Hello, ' + input;
```

File: modules/script/script-impl/src/test/resources/myapplication/inner-script.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+console.log("Inner script");
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapplication/inner-script.js:1]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
console.log('Inner script');
```

File: modules/script/script-impl/src/test/resources/myapplication/require-test.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+    console.log(JSON.stringify(app, null, 2));
+    console.log(JSON.stringify(xxx, null, 2));
+
+    var bean = __.newBean('com.enonic.xp.script.graal.GraalScriptBean');
+    bean.setSource("Source Name 1");
+    bean.execute();
+    bean.execute(1);
+    bean.execute([1,2,3]);
+    bean.execute("Hello", 1234);
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapplication/require-test.js:21]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
    bean.execute('Hello', 1234);
```

File: modules/script/script-impl/src/test/resources/myapplication/require-test.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
+const innerLib = require('inner-script');
+
+console.log('I am a require function');
+
+log.info("Print logs with two arguments: %s and %s ", 123.09, "456");
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/myapplication/require-test.js:5]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
log.info('Print logs with two arguments: %s and %s ', 123.09, "456");
```

File: modules/script/script-impl/src/test/resources/graaljs/function-script.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+input => {
```
Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/graaljs/function-script.js:1]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
(input) => {
```

Comment (alansemenov) [inline modules/script/script-impl/src/test/resources/graaljs/function-script.js:1]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=8692791)
```suggestion
(input) => {
```

**Review by anatol-sialitski — APPROVED** [review]

**Review by vbradnitski — APPROVED** [review]

**Pull request merged**